src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -101,7 +101,79 @@ public byte[] GetBytesFromBase64()
             {
                 throw ThrowHelper.GetFormatException(DateType.Base64String);
             }
+            return value;
+        }
+
+        /// <summary>
+        /// Parses the current JSON token value from the source as a <see cref="byte"/>.
+        /// Returns the value if the entire UTF-8 encoded token value can be successfully parsed to a <see cref="byte"/>
+        /// value.
+        /// Throws exceptions otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.Number"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <exception cref="FormatException">
+        /// Thrown if the JSON token value is either of incorrect numeric format (for example if it contains a decimal or 
+        /// is written in scientific notation) or, it represents a number less than <see cref="byte.MinValue"/> or greater 
+        /// than <see cref="byte.MaxValue"/>.
+        /// </exception>
+        public byte GetByte()
+        {
+            if (!TryGetByte(out byte value))
+            {
+                throw ThrowHelper.GetFormatException(NumericType.Byte);
```

Comment:
@ahsonkhan, we've probably discussed this previously, but I'm a little surprised we ended up with this throw pattern.  What's the rationale for `throw ThrowHelper.GetFormatException(...);` instead of `ThrowHelper.ThrowFormatException(...)`?

Reply:
I don't think we are consistent across the board here. I think, depending on who had written it/what they valued more at the time, they picked one or the other and the rationale for one or the other isn't super concrete (at least as far as I remember). I had used both myself (i.e. wasn't always consistent).

There are two patterns:
1) `throw HelperToGetException(...)`
   - Used to build up an exception with an exception message where building up the message is non-trivially expensive (or if there are multiple arguments and we need to avoid extra comparisons).
   - Not absolutely necessary, pattern 2 can be used just as well for most of these cases.
   - The benefit here is that the `throw` keyword is clearly visible in the caller (and so it's obvious that we aren't returning from this point).
2) `ThrowCallingHelperWhichThrows()`
   - Used when there isn't a single exception type that can be thrown, so a Helper that returns an exception wouldn't be feasible (however, you can also just return the base type `Exception` in pattern 1).
   - Moving the `throw` into the helper makes the caller code smaller.
   - Consistent with the existing pattern: 
https://github.com/dotnet/corefx/blob/fb81332b9643ecf8c0bb34d7938da97aa5e13e03/src/Common/src/CoreLib/System/ThrowHelper.cs#L86-L90
https://github.com/dotnet/corefx/blob/fb81332b9643ecf8c0bb34d7938da97aa5e13e03/src/System.Memory/src/System/ThrowHelper.cs#L27-L29

Ideally, we would just use pattern 2 anywhere it made sense to do so. In some cases, where the throwing is happening right at the top of a public method as part of argument validation, there is (arguably) _some_ value in keeping the `throw` keyword in the caller. That's probably why we have pattern 1 being used in this instance.

[sharplab.io for example](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdzJA2AQIkhgFlyACl48GCAJTHT5/7wAZgwuCAwAPAx0Xn7+ZibUcUkMGAAWUBAA7gxcMNkAklwK2JK8ACYA8mIy2HwQXACiCCpidVwuACQARAK4uNgA5jAgxggAvl1eGonJY7FxxADsngwApAz6ALybUdP+czNm8zpM5Na29k6kbh7evodxQSFhkdHz/gnJcWkZ2QDiMAwhWKpUq1SgtV49SaLTaoSm73MBy+SxW6y2OzoeyR82OljONjsDkc2huGE8Pk+ySeoQiURiDw+iKSABV0llgSVylUam0YTBWlD2t5sUlkclUWF0Qxtrt5uKjg8ANqOQGpCBlfLiSQuVVpDVasSSKptXAsAByEEKpS47gGXgAuvMxFBeMUMDBTtZOaCeRC+c0BW0GACgUUuWDeUL+YL6mSKfMqUlUbkCuHfeDIdDA7H2t1ev0hiMjONJqKGAqGM7Xe7PVYmCgGGzfj7uZmA7ChfHvInmSl2f9Aa3I/7ozm4SL5TQxkA===)